File: loginQrLine.py
import requests, json, time

hostVH = "https://api.vhtear.com/"
apikey = "PREMIUMKEY" #Chat me on whatsapp https://wa.me/+628123552767

#Header sesuai'in dengan scripmu
headerLine = {
  "User-Agent": "Line/6.3.2",
  "X-Line-Application": "DESKTOPWIN\t6.3.2\tVH-PC\t10.0.14;SECONDARY",
  "x-lal": "en_id"
}

def NewLogin():
  sagne = requests.post(hostVH+"zloginqr="+apikey, json=headerLine)
  if sagne.status_code == 200:
     tH = sagne.json()
     print("Link Login: "+tH["result"]["url"]) #Atau ganti dengan client.sendMessage(to,tH["result"]["url"])
  time.sleep(4)
  ve = requests.get(hostVH+"zgetcode="+apikey)
  if ve.status_code == 200:
    bG = ve.json()
    if bG["result"]["code"] != "":
        print("Pin code : ",bG["result"]["code"])
  time.sleep(4)
  ve = requests.get(hostVH+"zgettoken="+apikey)
  if ve.status_code == 200:
      rH = ve.json()
      if rH["result"]["token"] != "":
         print("Token : ",rH["result"]["token"]) #Atau ganti dengan client.sendMessage(to,rH["result"]["token"])
         print("Cert : ",rH["result"]["certificate"]) #Atau ganti dengan client.sendMessage(to,rH["result"]["certificate"])
         
  
def LoginWithCertificate(cert):
  sagne = requests.post(hostVH+"zloginqrwithcert="+apikey+"&"+cert, json=headerLine)
  if sagne.status_code == 200:
     ve = sagne.json()
     print("Link Login: "+ve["result"]["url"]) #Atau ganti dengan client.sendMessage(to,ve["result"]["url"])
  else:
     print("Error : ", sagne.status_code)
  time.sleep(4)
  ve = requests.get(hostVH+"zgettoken="+apikey)
  if ve.status_code == 200:
      cok = ve.json()
      if cok["result"]["token"] != "":
        print("Token : ",cok["result"]["token"]) #Atau ganti dengan client.sendMessage(to,cok["result"]["token"])
        
  
#LoginWithCertificate("certificate mu")
#NewLogin()

File: loginQrLine.go
package main

import (
	"fmt"
	"github.com/tidwall/gjson"
	"io/ioutil"
	"log"
	"net/http"
	"strings"
	"time"
)

var hosts = "https://api.vhtear.com/"
var apikey = "Premiumkey"
var headersSu = `{"User-Agent": "Line/6.3.2","X-Line-Application": "DESKTOPWIN\t6.3.2\tVH-PC\t10.0.14;SECONDARY","x-lal": "en_id"}`

func NewLoginV2() string {
	requestBody := strings.NewReader(headersSu)
	res, err := http.Post(hosts+"loginqr="+apikey, "application/json; charset=UTF-8", requestBody)
	if err != nil {
		log.Fatal(err)
	}
	data, _ := ioutil.ReadAll(res.Body)
	res.Body.Close()
	bodyString := string(data)
	get := gjson.Get(bodyString, "result.url")
	if get.Exists() {
		return get.String()
	} else {
		fmt.Println(bodyString)
		return ""
	}
}

func LoginWithCertificate(cert string) string {
	requestBody := strings.NewReader(headersSu)
	res, err := http.Post(hosts+"loginqrwithcert="+apikey+"&"+cert, "application/json; charset=UTF-8", requestBody)
	if err != nil {
		log.Fatal(err)
	}
	data, _ := ioutil.ReadAll(res.Body)
	res.Body.Close()
	bodyString := string(data)
	get := gjson.Get(bodyString, "result.url")
	if get.Exists() {
		return get.String()
	} else {
		fmt.Println(bodyString)
		return ""
	}
}

func GetCode() {
	for {
		time.Sleep(8 * time.Second)
		fmt.Println("Wait code")
		res, err := http.Get(hosts + "getcode=" + apikey)
		if err != nil {
			log.Fatal(err)
		}
		data, _ := ioutil.ReadAll(res.Body)
		res.Body.Close()
		//fmt.Println(string(data))
		get := gjson.Get(string(data), "result.code")
		if get.String() != "" {
			fmt.Println(get.String())
			break
		}
	}
}

func GetTokens() {
	for {
		time.Sleep(2 * time.Second)
		fmt.Println("Wait token")
		res, err := http.Get(hosts + "gettoken=" + apikey)
		if err != nil {
			log.Fatal(err)
		}
		data, _ := ioutil.ReadAll(res.Body)
		res.Body.Close()
		token := gjson.Get(string(data), "result.token")
		cert := gjson.Get(string(data), "result.certificate")
		if token.String() != "" {
			fmt.Println(token.String())
			fmt.Println(cert.String())
			break
		
	}
}

func main() {
	//fmt.Println(LoginWithCertificate("certMu"))
	//fmt.Println(NewLoginV2())
	//GetCode()
	//GetTokens()
}
